advanced/wallets/react-wallet-v2/src/lib/KernelSmartAccountLib.ts

```diff
@@ -0,0 +1,218 @@
+              } 
+        }).extend(bundlerActions)
+        //@ts-ignore
+        this.client = client
+        console.log('Smart account initialized',{address: account.address, chain: client.chain.name});
+    }
+
+    getMnemonic(): string {
+        throw new Error("Method not implemented.");
+    }
+    getPrivateKey(): string {
+        return this.#signerPrivateKey;
+    }
+    getAddress(): string {
+        if (!this.client) {
+            throw new Error("Client not initialized");
+        }
+        return this.client.account?.address || ''
+    }
+    async signMessage(message: string): Promise<string> {
+        if (!this.client) {
+            throw new Error("Client not initialized");
+        }
+        const signature = await this.client.account?.signMessage({message})
+        return signature || ''
+    }
+    async _signTypedData(domain: any, types: any, data: any, _primaryType?: string | undefined): Promise<string> {
+        if (!this.client) {
+            throw new Error("Client not initialized");
+        }
+        console.log('Signing typed data with Kernel Smart Account');
+        const primaryType = _primaryType || ''
+        const signature = await this.client.account?.signTypedData({ domain, types, primaryType, message: data })
+        return signature || ''
+    }
+    connect(provider: JsonRpcProvider): any {
+        if (!this.client) {
+            throw new Error("Client not initialized");
+        }
+        return this
+    }
+    async signTransaction(transaction: any): Promise<string> {
+        if (!this.client || !this.client.account) {
+            throw new Error("Client not initialized");
+        }
+        const signature = await this.client.account.signTransaction(transaction)
+        return signature || ''
+    }
+    async sendTransaction({
+        to,
+        value,
+        data
+      }: { to: Address; value: bigint; data: Hex }){
+        console.log('Sending transaction from smart account',{to,value,data})
+        if (!this.client || !this.client.account) {
+            throw new Error("Client not initialized");
+        }
+   
+        const txResult = await this.client.sendTransaction({
+            to,
+            value,
+            data,
+            account: this.client.account,
+            chain: this.chain
+        })
+        console.log('Transaction completed',{txResult});
+        
+        return txResult
+        
+    }
+
+    async issueSessionKey(address: `0x${string}`, permissions: string): Promise<string> {
+        if (!this.publicClient) {
+            throw new Error("Client not initialized");
+        }
+        if(!address){
+            throw new Error("Target address is required");
+        }
+        const parsedPermissions = JSON.parse(permissions)
+        const sessionKeyAddress = address
+        console.log('Issuing new session key',{sessionKeyAddress});
+        const emptySessionKeySigner = addressToEmptyAccount(sessionKeyAddress)
```

Comment:
curious: what's an `emptySmartAcconut`  ?

Reply:
Its an abstraction they made to indicate signer does not have private key and limited capabilities